### PR TITLE
plugin: login fix.

### LINKED ISSUE #52: plugin: log method it not able to print on cln log!

### PR DESCRIPTION
**Pull request by swaptr**
This will fix logging issue.
Fixes #52

### CODE REVIEW

**Comment by swaptr** [comment]
`2022-06-24T11:19:16.978Z UNUSUAL plugin-cln_plugin_class_example.exe: Killing plugin: Received a JSON-RPC response for non-existent request`
This runs into this error.

File: packages/cln_plugin/example/cln_plugin_class_example.dart [hunk 1]
```diff
@@ -29,6 +29,7 @@ class MyPlugin extends Plugin {
         usage: "",
         description: "This is an example of custom rpc method",
         callback: (plugin, request) => firstMethod(plugin, request));
+    log(level: "debug", message: "messageString");
```
Comment (vincenzopalazzo) [inline packages/cln_plugin/example/cln_plugin_class_example.dart:32]:
```suggestion
    log(level: "info", message: "messageString");
```

Try info :) not sure if exist `debug`

**Review by vincenzopalazzo — CHANGES_REQUESTED** [review]

**Comment by swaptr** [comment]
Nope, doesn't fix, same error.

**Review by vincenzopalazzo — CHANGES_REQUESTED** [review]
`stdout.write(jsonEncode(Request(id: 40, method:"log", params: payload).toJson()));`

Where the payload is

```
var payload = HashMap<String, Object>();
    payload["message"] = "blabla";
    payload["level"] = level;
```

We are sending a message to core lightning with a request, and tell "Hey please run a method called log and pass these params"

**Comment by vincenzopalazzo** [comment]
PR welcome in the rust version too 😄

File: packages/cln_plugin/lib/src/json_rpc/request.dart [hunk 2]
```diff
@@ -23,11 +23,9 @@ class Request {
   dynamic params;
 
   /// A constructor for creating a new class instance.
-  Request(
-      {this.id,
+  Request({this.id,
       required this.method,
-      required this.params,
-      this.jsonrpc = "2.0"});
```
Comment (vincenzopalazzo) [inline packages/cln_plugin/lib/src/json_rpc/request.dart:30]: Oh! this was never null :/

My bad!

**Comment by vincenzopalazzo** [comment]
There is a bug somewhere because this is the request that we sent

`"{\"jsonrpc\":\"2.0\",\"method\":\"log\",\"params\":{\"level\":\"info\",\"message\":\"calling custom rpc method\"}}"`

**Comment by vincenzopalazzo** [comment]
@swaptr Take care of this PR review and merge it :) was painful debugging the `toJson` method of the request object!

**Review by swaptr — COMMENTED** [review]
image
Works. Thanks to @vincenzopalazzo we finally have logging functionality. :fireworks:

**Review by vincenzopalazzo — APPROVED** [review]

**Pull request merged**